### DISCUSSION TITLE
Invoice Labels

### DISCUSSION THREAD

**Discussion by chrisonline** [kimai/kimai#3741] [Questions & Answers]
Is there a way to change the invoice labels?

**For example:**
`<th class="text-right text-nowrap">{{ 'label.amount'|trans }}</th>`

Want to change the `label.amount` for my invoice template I am creating.

**Comment by kevinpapst** [comment]
Just write `<th class="text-right text-nowrap">Anzahl</th>` to change it...

What is your question exactly? I don't get it.

**Reply by chrisonline** [reply]
Yes I can just write "Anzahl", but I want to create a template who is translatable.  
So I can possibly upload it as a new template to Github.

Is there a way to create a new invoice label?
And is it possible to change the current invoice label for example label.amount to "Betrag2" and in english to "Amount2".

**Reply by kevinpapst** [reply]
Aha, understand. You can create a plugin, which then allows to ship new translation keys, examples are included in  https://github.com/Keleo/DemoBundle/tree/1.x

You can change translation with your own bundle or you can buy the translation plugin, which allows to change most translations of the app directly from the Web UI: https://www.kimai.org/store/translation-bundle.html

But all of that prevents sharing the template in the [kimai invoice template repository](https://github.com/kimai/invoice-templates).

What translation exactly do you think is missing? Maybe is is generic enough to be included in Kimai directly...
The other option is simply a new variable that you define in the head of the template `{% set xyz = "My translation" %}` and then you use it later on `<th class="text-right text-nowrap">{{ xyz }}</th>`.

That allows for more scenarios like `if app.locale == 'en'  ... else ...` or template inheritance.

**Reply by chrisonline** [reply]
OK thanks for the infos.
I will check the possibilities you have listed.

For the missing translation:
Is there a place I can see ALL translations? 

Eventually the translation does already exists, but I don't know.
For example, I wanted to write instead of "Rechnungsbetrag" a simple "Gesamt" and for the "Rechnungsbetrag (netto)" "Zwischensumme". 
Where can I check if the translation "Gesamt" or "Zwischensumme" is already there?

**Reply by kevinpapst** [reply]
Grep through all `*.de.xlf` in `https://github.com/kimai/kimai/tree/main/translations` or use Weblate: https://hosted.weblate.org/projects/kimai/
